code/builders/builder-vite/src/index.ts

```diff
@@ -20,16 +20,17 @@ export { hasVitePlugins } from './utils/has-vite-plugins';
 
 export * from './types';
 
-function iframeMiddleware(options: Options, server: ViteDevServer): RequestHandler {
+function iframeMiddleware(options: Options, server: ViteDevServer): Polka.Middleware {
   return async (req, res, next) => {
-    if (!req.url.match(/^\/iframe\.html($|\?)/)) {
+    if (!req.url || !req.url.match(/^\/iframe\.html($|\?)/)) {
       next();
       return;
     }
+    const url = new URL(req.url, 'https://storybook.js.org');
```

Comment:
I'm guessing the base doesn't matter here, but it would probably be closer to reality if it was `http://localhost:6006`

Reply:
You're right. We should change it 👍